### PR TITLE
Util set project Command

### PR DESCRIPTION
**Pull request by obuchmann**
implements feature #3944

### CODE REVIEW

**Comment by LukeTowers** [comment]
I don't think this works. If it's set to required then won't it require that option even when the util command isn't set project? Also, what would the full command signature be to trigger this?

**Comment by obuchmann** [comment]
I'm sure, it works. Options are always optional. `InputOption::VALUE_REQUIRED` only means that if you specify that option an value has to be set via `=`, its no flag.

Of course you can specify the `projectId` option on all other util commands too. But they are not processed. This follows the way `debug` and `force` are implemented.

The signature is `artisan october:util set project --projectId=<id>`. I also added the signature as comment.

**Comment by w20k** [comment]
@LukeTowers , @obuchmann is correct on value_required.

```
InputOption::VALUE_REQUIRED
This value is required (e.g. --iterations=5 or -i5), the option itself is still optional;
```

It could be via `space` or `=`, see [docs](https://symfony.com/doc/master/components/console/console_arguments.html) for examples.

**Comment by LukeTowers** [comment]
Perfect, thanks for checking into that! Please document it here: https://github.com/octobercms/october/blob/master/modules/system/console/OctoberUtil.php#L29 and remove it from the method docblock.

File: modules/system/console/OctoberUtil.php [hunk 1]
```diff
@@ -328,4 +330,28 @@ protected function utilGitPull()
...
 
+    /**
+     * Set ProjectId from october.com
+     * Call via: artisan october:util set project --projectId=<id>
+     */
+    protected function utilSetProject(){
+
+        $projectId = $this->option('projectId');
+
+        if(null == $projectId){
```
Comment (LukeTowers) [inline modules/system/console/OctoberUtil.php]: please use `empty($projectId)` instead, and add a space after your `if`

File: modules/system/console/OctoberUtil.php [hunk 2]
```diff
@@ -328,4 +330,28 @@ protected function utilGitPull()
         }
     }
 
+    /**
+     * Set ProjectId from october.com
+     * Call via: artisan october:util set project --projectId=<id>
+     */
+    protected function utilSetProject(){
```
Comment (LukeTowers) [inline modules/system/console/OctoberUtil.php]: Opening brackets go on new lines for methods and classes

File: modules/system/console/OctoberUtil.php (hunk 1 above)
Comment (obuchmann) [reply modules/system/console/OctoberUtil.php]: resolved in https://github.com/octobercms/october/pull/3946/commits/32d9e14c2e3e2a4c15a3333325a1c6f33bc34c5b

File: modules/system/console/OctoberUtil.php [hunk 3]
```diff
@@ -328,4 +331,27 @@ protected function utilGitPull()
         }
     }
 
+    /**
```
Comment (LukeTowers) [inline modules/system/console/OctoberUtil.php]: remove this docblock

File: modules/system/console/OctoberUtil.php (hunk 2 above)
Comment (obuchmann) [reply modules/system/console/OctoberUtil.php]: resolved in https://github.com/octobercms/october/pull/3946/commits/32d9e14c2e3e2a4c15a3333325a1c6f33bc34c5b

File: modules/system/console/OctoberUtil.php (hunk 3 above)
Comment (obuchmann) [reply modules/system/console/OctoberUtil.php]: resolved in https://github.com/octobercms/october/pull/3946/commits/0680384f9a121b4e56a14d322f1f64723de238d2

**Pull request merged**